Uplift/Views/Headers/HomeScreenHeaderView.swift

```diff
@@ -28,29 +29,105 @@ class HomeScreenHeaderView: UIView {
         welcomeMessage.numberOfLines = 0
         welcomeMessage.text = getGreeting()
         addSubview(welcomeMessage)
-
+        
+        statusButton = StatusButton()
+        statusButton.layer.borderColor = UIColor.gray01.cgColor
+        statusButton.layer.borderWidth = 1
+        statusButton.layer.cornerRadius = 12
+        statusButton.addTarget(self, action: #selector(showCapacityView), for: .touchUpInside)
+        addSubview(statusButton)
+        
         setupLayout()
     }
-
+    
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
+    
+    @objc func showCapacityView() {
+        if (statusButton.tag == 0) {
```

Comment:
I feel like it would enhance readability to use a boolean variable instead of the tag property. Maybe as a property of `statusButton` like `statusButton.isDisclosed`.

Reply:
To add on, you could also do this in a way which moves a lot of functionality into the status button. This function could literally have two lines:
`statusButton.toggleDisclosure()`
`delegate.didToggleDisclosure() // For communicating with the homeviewcontroller`
and then everything else such as background color could stay in the status button